openpype/hosts/maya/api/lib.py

```diff
@@ -172,6 +174,218 @@ def maintained_selection():
+                json.dumps(preset, indent=4, sort_keys=True)
+            )
+        )
+    with ExitStack() as stack:
+        stack.enter_context(maintained_time())
+        stack.enter_context(panel_camera(
+            instance.data["panel"], preset["camera"]))
+        stack.enter_context(viewport_default_options(preset, instance))
+        if preset["viewport_options"].get("textures"):
+            stack.enter_context(material_loading_mode())
+            if preset["viewport_options"].get("reloadTextures"):
+                # Regenerate all UDIM tiles previews
+                reload_all_udim_tile_previews()
+        # not supported by `capture`
+        preset["viewport_options"].pop("reloadTextures", None)
+        path = capture.capture(log=self.log, **preset)
+        self.log.debug("playblast path  {}".format(path))
+
+    return path
+
+
+def generate_capture_preset(instance, camera, path,
+                            start=None, end=None, capture_preset=None):
+    """Function for getting all the data of preset options for
+    playblast capturing
+
+    Args:
+        instance (str): instance
+        camera (str): review camera
+        path (str): filepath
+        start (int): frameStart
+        end (int): frameEnd
+        capture_preset (dict): capture preset
+
+    Returns:
+        _type_: _description_
+    """
+    preset = load_capture_preset(data=capture_preset)
+
+    # "isolate_view" will already have been applied at creation, so we'll
+    # ignore it here.
+    preset.pop("isolate_view")
+
+    # Set resolution variables from capture presets
+    width_preset = capture_preset["Resolution"]["width"]
+    height_preset = capture_preset["Resolution"]["height"]
+
+    # Set resolution variables from asset values
+    asset_data = instance.data["assetEntity"]["data"]
+    asset_width = asset_data.get("resolutionWidth")
+    asset_height = asset_data.get("resolutionHeight")
+    review_instance_width = instance.data.get("review_width")
+    review_instance_height = instance.data.get("review_height")
+    preset["camera"] = camera
+
+    # Tests if project resolution is set,
+    # if it is a value other than zero, that value is
+    # used, if not then the asset resolution is
+    # used
+    if review_instance_width and review_instance_height:
+        preset["width"] = review_instance_width
+        preset["height"] = review_instance_height
+    elif width_preset and height_preset:
+        preset["width"] = width_preset
+        preset["height"] = height_preset
+    elif asset_width and asset_height:
+        preset["width"] = asset_width
+        preset["height"] = asset_height
+    preset["start_frame"] = start
+    preset["end_frame"] = end
+
+    # Enforce persisting camera depth of field
+    camera_options = preset.setdefault("camera_options", {})
+    camera_options["depthOfField"] = cmds.getAttr(
+        "{0}.depthOfField".format(camera))
+
+    preset["filename"] = path
+    preset["overwrite"] = True
```

Comment:
```suggestion
    preset = load_capture_preset(data=capture_preset)
    
    preset["camera"] = camera
    preset["start_frame"] = start
    preset["end_frame"] = end
    preset["filename"] = path
    preset["overwrite"] = True

    # "isolate_view" will already have been applied at creation, so we'll
    # ignore it here.
    preset.pop("isolate_view")

    # Set resolution variables from capture presets
    width_preset = capture_preset["Resolution"]["width"]
    height_preset = capture_preset["Resolution"]["height"]

    # Set resolution variables from asset values
    asset_data = instance.data["assetEntity"]["data"]
    asset_width = asset_data.get("resolutionWidth")
    asset_height = asset_data.get("resolutionHeight")
    review_instance_width = instance.data.get("review_width")
    review_instance_height = instance.data.get("review_height")

    # Tests if project resolution is set,
    # if it is a value other than zero, that value is
    # used, if not then the asset resolution is
    # used
    if review_instance_width and review_instance_height:
        preset["width"] = review_instance_width
        preset["height"] = review_instance_height
    elif width_preset and height_preset:
        preset["width"] = width_preset
        preset["height"] = height_preset
    elif asset_width and asset_height:
        preset["width"] = asset_width
        preset["height"] = asset_height

    # Enforce persisting camera depth of field
    camera_options = preset.setdefault("camera_options", {})
    camera_options["depthOfField"] = cmds.getAttr(
        "{0}.depthOfField".format(camera))
```
Let's move some simple attributes defined from function inputs or constants together to make things easier to read.

Reply:
Fixed in #6081